compiler/rustc_parse/messages.ftl

```diff
@@ -230,6 +230,10 @@ parse_expected_struct_field = expected one of `,`, `:`, or `{"}"}`, found `{$tok
 
 parse_expected_trait_in_trait_impl_found_type = expected a trait, found type
 
+parse_expr_rarrow_call = `->` used for field access or method call
+    .suggestion = try using `.` instead
+    .help = Rust uses `.` to access fields and methods
```

Comment:
ideally this message would explain that `.` already (mostly) explicitly derefs

Reply:
alternatively, if we expect this to be done mostly on raw pointers (`*mut`/`*const`), we could instead turn the structured suggestion into a multipart suggestion that wraps in `(*` + `).`

Reply:
> alternatively, if we expect this to be done mostly on raw pointers (*mut/*const), we could instead turn the structured suggestion into a multipart suggestion that wraps in (* + ).

This was not my intention. I expect the Rust beginner to discover we use `->` before they try to use `*mut`/`*const`.

I reworked the help message in order to include dereferences the value if needed. Please ping me if you feel this is not clear enough!